### PR TITLE
HPCC-27428 Switch to use changed_modules action

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: .github/workflows/build-test-eclwatch.yml [hunk 1]
```diff
@@ -1,15 +1,6 @@
 name: Build Test ECL Watch
 
 on:
-  push:
```
Comment (richardkchapman) [inline .github/workflows/build-test-eclwatch.yml:4]: Was this removed deliberately?

**Review by richardkchapman — COMMENTED** [review]
One question

File: .github/workflows/build-test-eclwatch.yml (hunk 1 above)
Comment (GordonSmith) [reply .github/workflows/build-test-eclwatch.yml:4]: Yes - there was no real need for it.

**Comment by GordonSmith** [comment]
@richardkchapman - I think I have spotted an edge case, so hold off for now.

**Comment by richardkchapman** [comment]
@GordonSmith I have forked the paths-filter repo into ours, in case you want to use it

The skip_check calculations are supposedly doing more than just checking paths though - they are also comparing against prior successful tests somehow

**Comment by GordonSmith** [comment]
Yes - the bit that is confusing me, is I get different results in the various workflow prejobs for the same PR...  It may be very very clever, but I am happy to continue using it for now.

**Review by richardkchapman — APPROVED** [review]
